tests/UnifiedSpecTests/Constraint/Matches.php

```diff
@@ -263,6 +268,35 @@ private function assertMatchesOperator(BSONDocument $operator, $actual, string $
             return;
         }
 
+        if ($name === '$$matchAsDocument') {
```

Comment:
I don't feel strongly about this since each if block returns. I'm not sure there's any technical difference, as [switch](https://www.php.net/manual/en/control-structures.switch.php) only discusses a benefit for complex conditions (not a variable reference) since it evaluates once before comparing with each `case`.

You're welcome to refactor this in a separate PR if you like, but modifying the entire method body in this PR would make the actual changes less visible.